apps/passport-server/src/services/generic-issuance/pipelines/CSVPipeline/makeGenericPODPCD.ts

```diff
@@ -0,0 +1,141 @@
+  CSVPipelineMatchConfig,
+  CSVPipelinePODEntryOptions
+} from "@pcd/passport-interface";
+import { ArgumentTypeName, SerializedPCD } from "@pcd/pcd-types";
+import { PODEntries } from "@pcd/pod";
+import { PODPCD, PODPCDPackage } from "@pcd/pod-pcd";
+import { v5 as uuidv5 } from "uuid";
+import { traced } from "../../../telemetryService";
+
+export async function makeGenericPODPCD(
+  inputRow: Record<string, string>,
+  eddsaPrivateKey: string,
+  requesterEmail: string | undefined,
+  requesterSemaphoreId: string | undefined,
+  pipelineId: string,
+  issueToUnmatchedEmail: boolean | undefined,
+  output: CSVPipelinePODEntryOptions,
+  matchConfig?: CSVPipelineMatchConfig
+): Promise<SerializedPCD<PODPCD> | undefined> {
+  return traced("", "makeGenericPODPCD", async () => {
+    if (!requesterEmail || !requesterSemaphoreId) {
+      return undefined;
+    }
+
+    // PODs can be issued either to anyone who makes a request, OR to anyone
+    // whose credential matches some of the values contained in the POD.
+    // To issue to anyone, `issueToUnmatchedEmail` must be enabled. If it is
+    // not, then some `matchConfig` must be specified.
+    if (issueToUnmatchedEmail !== true && !matchConfig) {
+      throw new Error("No configuration for matching requests to CSV data");
+    }
+
+    if (!issueToUnmatchedEmail) {
+      // If we are matching the user by comparing their email to some value in
+      // the CSV input data:
+      if (matchConfig?.type === "email") {
+        const emailColumnName = matchConfig.inputField;
+        // Check if the column exists
+        if (!(emailColumnName in inputRow)) {
+          throw new Error(
+            `Could not find column "${emailColumnName}" in CSV input data, could not match against user email address`
+          );
+        }
+        // If the value does not match, no POD is issued
+        if (inputRow[emailColumnName] !== requesterEmail) {
+          return undefined;
+        }
+      }
+      // If we are matching the user by comparing their Semaphore ID to some
+      // value in the CSV input data:
+      else if (matchConfig?.type === "semaphoreID") {
+        const semaphoreIDColumnName = matchConfig.inputField;
+        // Check if the column exists
+        if (!(semaphoreIDColumnName in inputRow)) {
+          throw new Error(
+            `Could not find column "${semaphoreIDColumnName}" in CSV input data, could not match against user Semaphore ID`
+          );
+        }
+        // If the value does not match, no POD is issued
+        if (inputRow[semaphoreIDColumnName] !== requesterSemaphoreId) {
+          return undefined;
+        }
+      }
+    }
+
+    const entries: PODEntries = {};
+
+    // Iterate over configured POD entries
+    for (const [name, entryConfig] of Object.entries(output)) {
+      switch (entryConfig.source.type) {
+        // This entry is populated from configuration
+        case "configured":
+          /* @todo non-string values */
```

Comment:
Can this use entryConfig.type in the same way as for "input" below?

Reply:
I can't see what you're getting at with that question.

Reply:
`entryConfig.type` seems to be used below to decide between POD types, so I wonder why you're not using that to handle non-string values here too.  I may be misunderstanding the input format and which inputs are available when.